kubespawner/spawner.py

```diff
@@ -2910,8 +3105,39 @@ async def _start(self):
             self._last_event = events[-1]["metadata"]["uid"]
 
         if self.storage_pvc_ensure:
-            pvc = self.get_pvc_manifest()
+            if (
+                self.handle_legacy_names
+                and not self._pvc_exists
+                and self._state_kubespawner_version == "unknown"
```

Comment:
add `and self.remember_pvc_name` because the result of this process won't be remembered otherwise, and note to this effect to help string.